test/Microsoft.Azure.Devices.Edge.Test/PlugAndPlay.cs

```diff
@@ -75,30 +82,71 @@ public async Task PlugAndPlayDeviceClient()
                 });
         }
 
+        [TestCase(Protocol.Mqtt, false)]
+        [TestCase(Protocol.Amqp, false)]
+        [TestCase(Protocol.Mqtt, true)]
+        [TestCase(Protocol.Amqp, true)]
         [Test]
-        public async Task PlugAndPlayModuleClient()
+        public async Task PlugAndPlayModuleClient(Protocol protocol, bool brokerOn)
         {
             CancellationToken token = this.TestToken;
             string loadGenImage = Context.Current.LoadGenImage.Expect(() => new ArgumentException("loadGenImage parameter is required for Priority Queues test"));
             EdgeDeployment deployment = await this.runtime.DeployConfigurationAsync(
                 builder =>
                 {
-                    builder.GetModule(ModuleName.EdgeHub).WithEnvironment(new[] { ("UpstreamProtocol", "Mqtt") });
+                    if (brokerOn)
+                    {
+                        this.AddBrokerToDeployment(builder);
+                    }
+
+                    builder.GetModule(ModuleName.EdgeHub).WithEnvironment(new[] { ("UpstreamProtocol", protocol.ToString()) });
                     builder.AddModule(LoadGenModuleName, loadGenImage)
                     .WithEnvironment(new[]
                     {
                             ("testStartDelay", "00:00:00"),
                             ("messageFrequency", "00:00:00.5"),
-                            ("transportType", Client.TransportType.Mqtt.ToString()),
+                            ("transportType", protocol.ToString()),
                             ("modelId", TestModelId)
                     });
                 },
                 token);
+
             EdgeModule filter = deployment.Modules[LoadGenModuleName];
             await filter.WaitForEventsReceivedAsync(deployment.StartTime, token);
             await this.ValidateModule(this.runtime.DeviceId, LoadGenModuleName, TestModelId);
         }
 
+        EdgeConfigBuilder AddBrokerToDeployment(EdgeConfigBuilder builder)
```

Comment:
this is interesting approach. Is there a way we can make it part of the e2e framework? Could be useful for everyone. Don't have to be done in this pr, but maybe the next one? //cc @damonbarry

Reply:
Probably the most clear way to do this (that I can think of) would be to add a version of this method to the runtime.DeployConfigurationAsync method. Then we can add an attribute with test data + parameter to every test (similar to this test) to give each test the option of turning this functionality on or not...
This is similar to how we deal with amqp or mqtt today

Another option is we could have an extra stage in the pipeline that runs all the tests again, but this time with a different deployment that has broker on. Requires no test code change, but I'm not sure how the test output would look